state-reconstruct-fetcher/src/types/v2.rs

```diff
@@ -0,0 +1,302 @@
+use ethers::{abi, types::U256};
+use serde::{Deserialize, Serialize};
+
+use super::ParseError;
+use crate::constants::zksync::{
+    L2_TO_L1_LOG_SERIALIZE_SIZE, LENGTH_BITS_OFFSET, OPERATION_BITMASK,
+};
+
+#[derive(Debug, Serialize, Deserialize)]
+pub enum PackingType {
+    Add,
+    Sub,
+    Transform,
+    NoCompression,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub enum L2ToL1Pubdata {
+    L2ToL1Log(Vec<u8>),
+    L2ToL2Message(Vec<u8>),
+    PublishedBytecode(Vec<u8>),
+    CompressedStateDiff {
+        /// NOTE: Does this make sense?
+        is_reapeated_write: bool,
```

Comment:
I'd say yes in this case, when it works, but there is a slight typo though :stuck_out_tongue: `is_reapeated_write` -> `is_repeated_write` :wink: